### DISCUSSION TITLE
LDAP authentification for official minio clients (mc or console).

### DISCUSSION THREAD

**Discussion by Lumiond** [minio/minio#11691] [Q&A] [answered]
Hello! I have asked this question in slack but nobody hasn't answered.

We need to use AD for the users authentification. I set
```
MINIO_IDENTITY_LDAP_SERVER_ADDR (my AD IP addr)
MINIO_IDENTITY_LDAP_USERNAME_FORMAT (cn=%s,cn=Users,dc=test,dc=local)
MINIO_IDENTITY_LDAP_STS_EXPIRY (1h)
MINIO_IDENTITY_LDAP_TLS_SKIP_VERIFY (on)
MINIO_IDENTITY_LDAP_SERVER_INSECURE (on)
```
and after this only example go program from ldap manual returns keys and token (not in pretty format like described in manual), on POST request response is "access denied" and also I can't use even mc for filling bucket (we make some tests now and we need to put some info to buckets using AD users). How can I login with AD credentials and put/delete some files into storage through mc, webUI or console? Do we need to write our own client for our tasks? Minio version is 2021-02-24T18:44:45Z

PS: sorry for my English, it's not my native language.

**Comment by harshavardhana** [comment, marked as answer]
LDAP authentication already works with minio console all you need to do is `CONSOLE_LDAP_ENABLED`  with `mc` you need to use the STS API to get the credentials and use it with `mc` 

https://github.com/minio/mc/blob/master/docs/minio-client-complete-guide.md#rotating-credentials
